### DISCUSSION TITLE
Could I get your opinion on something....

### DISCUSSION THREAD

**Discussion by PAEz** [Picorims/wav2bar#65] [Q&A]
Hey, I tried your app and making the effect was fun but when I tried to render it the render looked completely different (size mainly).  This has lead me to want to make my own (I dont even need a ui, coz I cant do that ;P ) and was wondering if you could comment on is this a good idea.....I dont have much knowledge and less experience on this stuff.

Im going to target 60 frames a second, you can render a video at less than 60 but it will still calculate all the intermediate frames.  This is to make some effects like trails using a double buffer work right amongst other things.
When loading in a sound file it will process the fft for all the frames and store the results.  These results are what the effect is processed from.  This will allow me to do multiple tracks at once with lots of speed hopefully.  Using multiple tracks that are made with a AI stem separator allows to get much more reactive videos.  It also allows me to collect other data such as what was the highest value for a frequency, the averages, etc.  Also makes reloads HEAPS faster, no need to process again.  Ill just stick all project files in a zip probably, done that before.  And I want all of this in just the browser hopefully.  Seen I can create video in the browser, just havent seen it with sound yet, but since they did ffmpeg in wasm then Im sure its in there.  And for a learning thing and cause I think itll be great I want to use Pixi for the graphics.  Fast (and the next version is even faster!) and with heaps of good features.  Plus its easy to take screenshots of.  Noticed you used html2img which can have problems, kinda just shoulda gone for svg huh?

And lastly your on gpl, can I take the smallest amount of code out of this without being sucked into the gpl vacuum?   I just need the bit to process the fft offline (not real time).  I cant find another example ANYWHERE!!! and Im mainly a copy and paste coder (limited mental ability that is decreasing).

Good luck on the new version.  Tauri is rather cool.

PS....bahahahaha, just looked at your new code and noticed your using Pixi, NICE!!!!!  Maybe I should just wait to see what you do now ;P

**Comment by Picorims** [comment]
Hi,

The difference between the rendering of both is a known issue due to the difference of rendering methods between preview and export. It should be limited to particle and visualizers movement behavior exclusively though.

Well it is up to you, just be ready to face the potential synchronization issues that I faced a lot. Decoupling rendering and computation is a good idea and how it should have been done. For fft caching though beware that it could easily fill up memory depending of the track length, depending of how it is done. Putting it into the save isn't a bad idea either. You can try doing it all in the browser if you can execute ffmpeg commands but again, you might face memory issues. Pixi is a good pick, and yes doing a screenshot was a dirty hack and workaround, html is not a good choice at all.

For licensing it depends which parts, though I don't think you need it that much since you have the principle and all those steps can be found online. It is not well written. Could you point me to specific areas? I understand the concern about gpl and the goal is that it stays open source, hence the choice. It does not prevent taking inspiration of the (bad) architecture.

As for the new version, it is just some prototyping, it is unlikely that I rewrite it, let alone finish the prototype due to personal reasons.

Let me know if you have questions,

~ Picorims
